src/eventsource.js

```diff
@@ -56,7 +56,22 @@ $.EventSource = function() {
 /** @lends OpenSeadragon.EventSource.prototype */
 $.EventSource.prototype = {
 
-    // TODO: Add a method 'one' which automatically unbinds a listener after the first triggered event that matches.
+    /**
+     * Add an event handler to be triggered only once for a given event.
+     * @function
+     * @param {String} eventName - Name of event to register.
+     * @param {OpenSeadragon.EventHandler} handler - Function to call when event
+     * is triggered.
+     * @param {Object} [userData=null] - Arbitrary object to be passed unchanged
+     * to the handler.
+     */
+    addOnceHandler: function(eventName, handler, userData) {
+        var self = this;
+        this.addHandler(eventName, function onceHandler(event) {
```

Comment:
I think we found that on some browser it didn't like having the function named like that. Instead we had to put the new function in a var and pass that. Unfortunately, I don't remember the exact incident, but maybe it's worth doing that way anyway?